ale_linters/php/phpstan.vim

```diff
@@ -0,0 +1,39 @@
+" Author: medains <https://github.com/medains>, ardis <https://github.com/ardisdreelath>
+" Description: phpstan for PHP files
+
+" Set to change the ruleset
+let g:ale_php_phpstan_executable = get(g:, 'ale_php_phpstan_executable', 'phpstan')
+let g:ale_php_phpstan_level = get(g:, 'ale_php_phpstan_level', '4')
+
+function! ale_linters#php#phpstan#GetCommand(buffer) abort
+    return ale#Var(a:buffer, 'php_phpstan_executable')
+    \   . ' analyze -l'
+    \   . ale#Var(a:buffer, 'php_phpstan_level')
+    \   . ' --errorFormat raw'
+    \   . ' %s'
+endfunction
+
+function! ale_linters#php#phpstan#Handle(buffer, lines) abort
+    " Matches against lines like the following:
+    "
+    " filename.php:15:message
+    let l:pattern = '^.*\.php:\(\d\+\):\(.\+\)$'
```

Comment:
I just realised. I used to work at a company where we named some files `.inc` if they were library files to be included in other files. You might want to use a regex where the file extension doesn't matter.

I recommend looking at the regex used for some of the C linters. You can write a regex which looks for a possible Windows drive letter at the start, and then anything except `:` after that, up to the `:` for the line number.